great_expectations/experimental/datasources/sql_datasource.py

```diff
@@ -138,6 +134,158 @@ def test_connection(self, table_asset: TableAsset) -> None:
+        return ["year"]
+
+    def splitter_method_kwargs(self) -> Dict[str, Any]:
+        return {"column_name": self.column_name}
+
+    def request_options_to_batch_spec_kwarg_identifiers(
+        self, options: BatchRequestOptions
+    ) -> Dict[str, Any]:
+        return _request_options_to_datetime_batch_spec_kwargs(self, options)
+
+
+class ColumnSplitterYearAndMonth(_ColumnSplitter):
+    method_name: Literal["split_on_year_and_month"] = "split_on_year_and_month"
+
+    @property
+    def param_names(self) -> List[str]:
+        return ["year", "month"]
+
+    def splitter_method_kwargs(self) -> Dict[str, Any]:
+        return {"column_name": self.column_name}
+
+    def request_options_to_batch_spec_kwarg_identifiers(
+        self, options: BatchRequestOptions
+    ) -> Dict[str, Any]:
+        return _request_options_to_datetime_batch_spec_kwargs(self, options)
+
+
+class ColumnSplitterYearAndMonthAndDay(_ColumnSplitter):
+    method_name: Literal[
+        "split_on_year_and_month_and_day"
+    ] = "split_on_year_and_month_and_day"
+
+    @property
+    def param_names(self) -> List[str]:
+        return ["year", "month", "day"]
+
+    def splitter_method_kwargs(self) -> Dict[str, Any]:
+        return {"column_name": self.column_name}
+
+    def request_options_to_batch_spec_kwarg_identifiers(
+        self, options: BatchRequestOptions
+    ) -> Dict[str, Any]:
+        return _request_options_to_datetime_batch_spec_kwargs(self, options)
+
+
+class ColumnSplitterDatetimePart(_ColumnSplitter):
+    datetime_parts: List[str]
+    column_name: str
+    method_name: Literal["split_on_date_parts"] = "split_on_date_parts"
+
+    @property
+    def param_names(self) -> List[str]:
+        return self.datetime_parts
+
+    def splitter_method_kwargs(self) -> Dict[str, Any]:
+        return {"column_name": self.column_name, "date_parts": self.param_names}
+
+    def request_options_to_batch_spec_kwarg_identifiers(
+        self, options: BatchRequestOptions
+    ) -> Dict[str, Any]:
+        return _request_options_to_datetime_batch_spec_kwargs(self, options)
+
+    @pydantic.validator("datetime_parts", each_item=True)
+    def _check_param_name_allowed(cls, v: str):
+        from great_expectations.execution_engine.split_and_sample.data_splitter import (
+            DatePart,
+        )
+
+        allowed_date_parts = [part.value for part in DatePart]
+        assert (
+            v in allowed_date_parts
+        ), f"Only the following param_names are allowed: {allowed_date_parts}"
+        return v
+
+
+DatetimeColumnSplitter = Union[
+    ColumnSplitterYear,
+    ColumnSplitterYearAndMonth,
+    ColumnSplitterYearAndMonthAndDay,
+    ColumnSplitterDatetimePart,
+]
+
+
+def _request_options_to_datetime_batch_spec_kwargs(
+    splitter: DatetimeColumnSplitter, options: BatchRequestOptions
+) -> Dict[str, Any]:
+    identifiers: Dict = {}
+    for part in splitter.param_names:
+        if part not in options:
```

Comment:
@billdirks As a small potential optimization, there may be a way to ensure at the top of this method that the parts (i.e., `splitter.param.names`) are a subset of options and point out the outliers (we have code like that in `ExecutionEngine` implementations).  Thanks.

Reply:
I am not sure this would be an optimization. It would slow it down in the case where it is a subset which is likely the most common use case since it is configured once then run multiple times. In the case where is it is not a subset, it prevents a handful (at most 5) dictionary assignments.